controllers/confidential_handler.go

```diff
@@ -0,0 +1,71 @@
+package controllers
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func patchConfigMap(client client.Client, name string, namespace string, data map[string]string) error {
+	configMap := &corev1.ConfigMap{}
+	if err := client.Get(context.TODO(), types.NamespacedName{
+		Name:      name,
+		Namespace: namespace,
+	}, configMap); err != nil {
+		return err
+	}
+
+	for key, value := range data {
+		configMap.Data[key] = value
+	}
+
+	return client.Update(context.TODO(), configMap)
+}
+
+func (r *KataConfigOpenShiftReconciler) toggleFeatureConfidential(enabled bool) error {
+
+	var ImageConfigMapData map[string]string
```

Comment:
nit: s/ImageConfigMapData/imageConfigMapData
since this is not an exported var

Reply:
Also pls change `enabled bool`  to `state FeatureGateState` to be consistent. 
While today we just have two states for the featuregate (fg), so bool is ok, but having the fg state as enum will retain the flexibility in case we want to introduce some additional states.